plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/utils/PlayAssetDeliveryUtilsTest.java

```diff
@@ -0,0 +1,63 @@
+package com.google.play.core.godot.assetpacks.utils;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+
+import org.godotengine.godot.Dictionary;
+import org.junit.Test;
+
+public class PlayAssetDeliveryUtilsTest {
+  private Dictionary constructAssetPackStateDictionaryHelper(
+      long bytesDownloaded,
+      int errorCode,
+      String name,
+      int status,
+      long totalBytesToDownload,
+      int transferProgressPercentage) {
+    Dictionary returnDict = new Dictionary();
+    returnDict.put("bytesDownloaded", bytesDownloaded);
+    returnDict.put("errorCode", errorCode);
+    returnDict.put("name", name);
+    returnDict.put("status", status);
+    returnDict.put("totalBytesToDownload", totalBytesToDownload);
+    returnDict.put("transferProgressPercentage", transferProgressPercentage);
+    return returnDict;
+  }
+
+  @Test
+  public void testAssetPackStateSerialization1() {
+    Dictionary testDictionary =
+        constructAssetPackStateDictionaryHelper(42, 0, "awesomePack", 2, 65536, 35);
+    AssetPackState testAssetPackState =
+        PlayAssetDeliveryUtils.convertDictionaryToAssetPackState(testDictionary);
+    Dictionary resultingDictionary =
+        PlayAssetDeliveryUtils.convertAssetPackStateToDictionary(testAssetPackState);
+    assertThat(resultingDictionary).isEqualTo(testDictionary);
+  }
+
+  @Test
+  public void testAssetPackStateSerialization2() {
+    Dictionary testDictionary =
+        constructAssetPackStateDictionaryHelper(
+            0, -6, "Lorem ipsum dolor sit amet, consectetur adipiscing elit.", 7, 0, 0);
+    AssetPackState testAssetPackState =
+        PlayAssetDeliveryUtils.convertDictionaryToAssetPackState(testDictionary);
+    Dictionary resultingDictionary =
+        PlayAssetDeliveryUtils.convertAssetPackStateToDictionary(testAssetPackState);
+    assertThat(resultingDictionary).isEqualTo(testDictionary);
+  }
+
+  @Test
+  public void testAssetPackStateSerialization3() {
+    // Test failure case where there is a missing key
+    Dictionary testDictionary =
+        constructAssetPackStateDictionaryHelper(42, 0, "awesomePack", 2, 65536, 35);
+    testDictionary.remove("bytesDownloaded");
+    AssetPackState testAssetPackState =
```

Comment:
I would adjust this test case to ensure that the result of `convertDictionaryToAssetPackState` is null or throws an exception, rather than checking after `convertAssetPackStateToDictionary `.

Reply:
Adjusted the test case. Currently only ```convertDictionaryToAssetPackState()``` can return null, we we are asserting isEqualTo(null) for that function call.